File: codegen/results/v001/qwen3.5_27b-nvfp4_nothink/20260408_153018/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	for i := 3; i*i <= n; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	if len(os.Args) != 2 {
		fmt.Fprintf(os.Stderr, "Usage: %s <integer N>\n", os.Args[0])
		os.Exit(1)
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: invalid integer argument\n")
		os.Exit(1)
	}

	if n < 2 {
		// No primes to print
		return
	}

	for i := 2; i <= n; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}

File: codegen/results/v001/qwen3.5_27b-nvfp4_nothink/20260408_153018/python/008_prime_numbers/solution.py
import sys
import math

def sieve_of_eratosthenes(limit):
    if limit < 2:
        return []
    
    # Create a boolean array "is_prime[0..limit]" and initialize all entries as true.
    is_prime = [True] * (limit + 1)
    is_prime[0] = is_prime[1] = False
    
    for p in range(2, int(math.isqrt(limit)) + 1):
        if is_prime[p]:
            # Mark multiples of p starting from p*p as not prime
            for i in range(p * p, limit + 1, p):
                is_prime[i] = False
    
    return [i for i in range(2, limit + 1) if is_prime[i]]

def main():
    if len(sys.argv) != 2:
        print("Usage: python script.py <integer>")
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Error: Argument must be an integer.")
        sys.exit(1)

    primes = sieve_of_eratosthenes(n)
    
    for prime in primes:
        print(prime)

if __name__ == "__main__":
    main()